src/AbstractHyperrectangle.jl

```diff
@@ -128,19 +128,52 @@ Return the list of vertices of a hyperrectangular set.
 ### Output
 
 A list of vertices.
+Zeros in the radius are correctly handled, i.e., the result does not contain any
+duplicate vertices.
 
 ### Notes
 
 For high dimensions, it is preferable to develop a `vertex_iterator` approach.
 """
 function vertices_list(H::AbstractHyperrectangle{N}
                       )::Vector{Vector{N}} where {N<:Real}
-    # fast evaluation if H has radius 0
-    if radius_hyperrectangle(H) == zeros(N, dim(H))
-        return [center(H)]
+    n = dim(H)
+
+    # three-valued vector; initially 0 if radius is zero and 1 otherwise
+    trivector = Vector{Int8}(undef, n)
+    m = 1
+    c = center(H)
+    v = copy(c)
+    @inbounds for i in 1:n
+        ri = radius_hyperrectangle(H, i)
+        if iszero(ri)
+            trivector[i] = Int8(0)
+        else
+            v[i] += ri
+            trivector[i] = Int8(1)
+            m *= 2
+        end
+    end
+
+    # create vertices by modifying the three-valued vector and constructing the
```

Comment:
Can you pull this information to a new `Algorithm` section? And maybe extend it, explaining the purpose of the three-valued vector.

Reply:
Indeed I had to think about it myself again :smile: